src/pages/home/report/ReportActionCompose.js

```diff
@@ -230,6 +231,75 @@ class ReportActionCompose extends React.Component {
         return this.props.translate('reportActionCompose.writeSomething');
     }
 
+    /**
+     * Returns the list of report compose actions
+     *
+     * @param {Boolean} showSplitBill
+     * @param {Boolean} showSendRequestMoney
+     * @param {Boolean} hasExcludedIOUEmails
+     * @param {Function} openPicker
+     * @param {Function} displayFileInModal
+     * @returns {Array<object>}
+     */
+    getReportComposeActions(showSplitBill, showSendRequestMoney, hasExcludedIOUEmails, openPicker, displayFileInModal) {
+        return [
+            ...(!hasExcludedIOUEmails
+                && Permissions.canUseIOU(this.props.betas)
+                && showSplitBill ? [
+                    {
+                        icon: Expensicons.Receipt,
+                        text: this.props.translate('iou.splitBill'),
+                        onSelected: () => {
+                            Navigation.navigate(
+                                ROUTES.getIouSplitRoute(
+                                    this.props.reportID,
+                                ),
+                            );
+                        },
+                    },
+                ] : []),
+            ...(!hasExcludedIOUEmails
+                && Permissions.canUseIOU(this.props.betas)
+                && showSendRequestMoney ? [
+                    {
+                        icon: Expensicons.MoneyCircle,
+                        text: this.props.translate('iou.requestMoney'),
+                        onSelected: () => {
+                            Navigation.navigate(
+                                ROUTES.getIouRequestRoute(
+                                    this.props.reportID,
+                                ),
+                            );
+                        },
+                    },
+                ] : []),
+            ...(!hasExcludedIOUEmails && Permissions.canUseIOUSend(this.props.betas) && showSendRequestMoney ? [
+                {
+                    icon: Expensicons.Send,
+                    text: this.props.translate('iou.sendMoney'),
+                    onSelected: () => {
+                        Navigation.navigate(
+                            ROUTES.getIOUSendRoute(
+                                this.props.reportID,
+                            ),
+                        );
+                    },
+                },
+            ] : []),
+            {
+                icon: Expensicons.Paperclip,
+                text: this.props.translate('reportActionCompose.addAttachment'),
+                onSelected: () => {
+                    openPicker({
+                        onPicked: (file) => {
+                            displayFileInModal({file});
+                        },
+                    });
+                },
+            },
+        ];
```

Comment:
Code is not DRY, we have redundant checks here.  Can you refactor this?

Reply:
Refactoring this might result into nested ternary operators. I was discouraged to use nested ternary operators in previous PRs. Any suggestion?

Reply:
We could use simple if/else no need for ternary here.

Reply:
It will be something like this. Any suggestion?
```
        const actions = [];
        if (!hasExcludedIOUEmails) {
            if (Permissions.canUseIOU(this.props.betas)) {
                if (showSplitBill) {
                    actions.push(
                        {
                            icon: Expensicons.Receipt,
                            text: this.props.translate('iou.splitBill'),
                            onSelected: () => {
                                Navigation.navigate(
                                    ROUTES.getIouSplitRoute(
                                        this.props.reportID,
                                    ),
                                );
                            },
                        },
                    );
                }
                if (showSendRequestMoney) {
                    actions.push({
                        icon: Expensicons.MoneyCircle,
                        text: this.props.translate('iou.requestMoney'),
                        onSelected: () => {
                            Navigation.navigate(
                                ROUTES.getIouRequestRoute(
                                    this.props.reportID,
                                ),
                            );
                        },
                    });
                }
            }
            if (Permissions.canUseIOUSend(this.props.betas) && showSendRequestMoney) {
                actions.push({
                    icon: Expensicons.Send,
                    text: this.props.translate('iou.sendMoney'),
                    onSelected: () => {
                        Navigation.navigate(
                            ROUTES.getIOUSendRoute(
                                this.props.reportID,
                            ),
                        );
                    },
                });
            }
        }

        actions.push({
            icon: Expensicons.Paperclip,
            text: this.props.translate('reportActionCompose.addAttachment'),
            onSelected: () => {
                openPicker({
                    onPicked: (file) => {
                        displayFileInModal({file});
                    },
                });
            },
        });
        return actions;
```

Reply:
Let's just move IOU options computation alone to a new method, leave Add Attachment untouched thanks!

```js
 getIOUOptions(reportParticipants) {
        const iouOptions = [];
        const participants = _.filter(reportParticipants, email => this.props.myPersonalDetails.login !== email).length > 1;
        const hasExcludedIOUEmails = lodashIntersection(reportParticipants, CONST.EXPENSIFY_EMAILS).length > 0;
        const hasMultipleParticipants = participants.length > 1;

        if (hasExcludedIOUEmails || participants.length === 0 || Permissions.canUseIOU(this.props.betas)) {
            return iouOptions;
        }

        if (hasMultipleParticipants) {
            // append split
        }
        else {
            if (Permissions.canUseIOUSend(this.props.betas)) {
                // append send
            }
            // append request
        }

        return iouOptions;
    }
```

cc: @iwiznia

Reply:
Done.